lib/model.js

```diff
@@ -3085,14 +3086,23 @@ class Model {
 
             // Convert date fields to real date objects
             if (this.constructor._hasDateAttributes && this.constructor._isDateAttribute(key) && !!value && !value._isSequelizeMethod) {
-              if (!(value instanceof Date)) {
-                value = new Date(value);
-              }
-              if (!(originalValue instanceof Date)) {
-                originalValue = new Date(originalValue);
-              }
-              if (originalValue && value.getTime() === originalValue.getTime()) {
-                return this;
+              // Dont parse DATEONLY to new Date, keep them as string
+              if (this.rawAttributes[key].type instanceof DataTypes.DATEONLY) {
+                if (originalValue && originalValue === value) {
+                  return this;
+                } else {
+                  value = moment(value).format('YYYY-MM-DD');
```

Comment:
Hm... I don't know, do we have other cases where values passed to `build`/`set` are converted to another data type? Might be unexpected behavior, I would think that these two are equivalent:
```js
const user = new User({birthday: new Date()})
const user = Object.assign(new User(), {birthday: new Date()});